iOSClient/Viewer/NCViewerMedia/NCViewerMediaDetailView.swift

```diff
@@ -112,101 +96,110 @@ class NCViewerMediaDetailView: UIView {
+        sizeLabel.text = CCUtility.transformedSize(metadata.size)
 
-            dateLabel.text = NSLocalizedString("_date_", comment: "")
-            dateValue.text = dateString
-        } else {
-            dateLabel.text = NSLocalizedString("_date_", comment: "")
-            dateValue.text = NSLocalizedString("_not_available_", comment: "")
+        if let shutterSpeedApex = exif.shutterSpeedApex {
+            shutterSpeedLabel.text = "1/\(Int(pow(2, shutterSpeedApex))) s"
         }
-        dateValue.textColor = textColor
 
-        // Dimension
-        if let image = image {
-            dimLabel.text = NSLocalizedString("_resolution_", comment: "")
-            dimValue.text = "\(Int(image.size.width)) x \(Int(image.size.height))"
+        if let iso = exif.iso {
+            isoLabel.text = "ISO \(iso)"
         }
-        dimValue.textColor = textColor
 
-        // Model
-        if let lensModel = lensModel {
-            lensModelLabel.text = NSLocalizedString("_model_", comment: "")
-            lensModelValue.text = lensModel
-            lensModelValue.textColor = textColor
+        if let apertureValue = exif.apertureValue {
+            apertureLabel.text = "ƒ\(apertureValue)"
         }
 
-        // Message
-        if metadata.isImage && !CCUtility.fileProviderStorageExists(metadata) && metadata.session.isEmpty {
-            messageButton.setTitle(NSLocalizedString("_try_download_full_resolution_", comment: ""), for: .normal)
-            messageButton.isHidden = false
-        } else {
-            messageButton.setTitle("", for: .normal)
-            messageButton.isHidden = true
+        if let exposureValue = exif.exposureValue {
+            exposureValueLabel.text = "\(exposureValue) ev"
+        }
+
+        if let lensLength = exif.lensLength {
+            lensSizeLabel.text = "\(lensLength) mm"
         }
 
-        // Location
-        if let location = location {
-            locationButton.setTitle(location, for: .normal)
-            locationButton.isHidden = false
+        if let date = exif.date {
+            let formatter = DateFormatter()
+
+            formatter.dateFormat = "EEEE"
+            let dayString = formatter.string(from: date as Date)
+            dayLabel.text = dayString
+
+            formatter.dateFormat = "d MMM yyyy"
+            let dateString = formatter.string(from: date as Date)
+            dateLabel.text = dateString
+
+            formatter.dateFormat = "HH:mm"
+            let timeString = formatter.string(from: date as Date)
+            timeLabel.text = timeString
         } else {
-            locationButton.setTitle("", for: .normal)
-            locationButton.isHidden = true
+            dayLabel.text = NSLocalizedString("_no_day_", comment: "")
+            dateLabel.text = NSLocalizedString("_no_date_", comment: "")
+            timeLabel.text = NSLocalizedString("_no_time_", comment: "")
```

Comment:
I think we can use `No timestamp available` in these cases which will align with other strings, but yes this will demand storyboard changes as well ig.